recorder/record.py

```diff
@@ -1,105 +1,59 @@
-
-    def save_streamed_response(self):
-        start_time = time.time()
-        filename = self.get_filename()
-        file = self.open_blob(filename)
-        chunk_size = 512 * 1024 # 0.5MB
-        attempts = 1
-        recording = True
-
-        while recording:
-            try:
-                logging.info(f'Attempt #{attempts}')
-                with requests.get(self.STREAM_URL, stream=True) as response:
-                    response.raise_for_status()
-                    logging.info(f'{filename} - starting ...')                       
-                    for chunk in response.iter_content(chunk_size=chunk_size):
-                        if chunk:
-                            file.write(chunk)
-                        else:
-                            logging.info('No chunk to write')
-                        
-                        elapsed_time = time.time() - start_time
-                        self.log_progress(filename, elapsed_time)         
-                        if elapsed_time >= self.duration:
-                            recording = False
-                            break
-            except Exception:                
-                logging.exception('Exception occurred')
-                attempts += 1
-                if attempts <= self.MAX_ATTEMPTS:                    
-                    time.sleep(attempts)
-                    continue
-                else:
-                    logging.critical(self.MAX_ATTEMPTS_EXCEEDED)
-                    sys.exit(1)
-
-if __name__ == '__main__':    
-    r = Recorder()
-    r.save_streamed_response()
+import signal
+import google.cloud.logging
+from datetime import datetime
+from google.cloud import storage
+from http.client import HTTPConnection
+from zoneinfo import ZoneInfo
+
+# allll the logging
+HTTPConnection.debuglevel = 1
+logging.basicConfig()
+logging.getLogger().setLevel(logging.DEBUG)
+requests_log = logging.getLogger("urllib3")
+requests_log.setLevel(logging.DEBUG)
+requests_log.propagate = True
+lc = google.cloud.logging.Client()
+lc.setup_logging()
+
+# handle timeouts
+class TimeoutException(Exception):
+    pass
+
+def timeout_handler(signum, frame):
+    raise TimeoutException
+
+signal.signal(signal.SIGALRM, timeout_handler)
+
+
+def record():
+    sc = storage.Client()
+    bucket = sc.bucket(os.getenv('BUCKET_NAME'))
+
+    r = requests.get('http://peridot.streamguys.com:5180/live', stream=True, timeout=25)
+    r.raise_for_status()
+    size = 320 * 1024 # ~20 seconds of audio
+    
+    for chunk in r.iter_content(chunk_size=size):
+        # reset timer
+        signal.alarm(0)
+
+        now = datetime.now(ZoneInfo('America/Chicago'))
+        ts = now.strftime('%w-%-H-%M-%S-%Y-%m-%d')
+        name = f'WCXP-LP-{ts}.mp3'
+        blob = bucket.blob(name)
+        writer = blob.open('wb')
+        writer.write(chunk)
+        writer.close()
+        logging.info(f'wrote {name}')
+
+        # start timer
+        signal.alarm(30)
```

Comment:
If the connection times out or the stream hangs for too long, this alarm raises an exception.